validator/sawtooth_validator/execution/executor.py

```diff
@@ -13,29 +13,41 @@
 # limitations under the License.
 # ------------------------------------------------------------------------------
 
+import json
 import logging
 import threading
+import queue
+
 
 from sawtooth_validator.protobuf import processor_pb2
 from sawtooth_validator.protobuf import transaction_pb2
 from sawtooth_validator.protobuf import validator_pb2
 
 from sawtooth_validator.execution.scheduler_serial import SerialScheduler
 from sawtooth_validator.execution import processor_iterator
+from sawtooth_validator.state import config_view
+from sawtooth_validator.state.state_view import StateViewFactory
 
 
 LOGGER = logging.getLogger(__name__)
 
 
 class TransactionExecutorThread(threading.Thread):
-    def __init__(self, service, context_manager, scheduler, processors,
-                 require_txn_processors=False):
+    def __init__(self,
+                 service,
+                 context_manager,
+                 scheduler,
+                 processors,
+                 state_database):
```

Comment:
Don't pass the state database, base the config_view_factory.  We want to limit components to read access unless they really need it.